src/user/user.service.ts

```diff
@@ -261,7 +264,9 @@ export class UserService {
     }
   }
 
-  async ban(req: any, input: BanDto) {
+  async ban(req: Request, input: BanDto) {
+    const objUser = req.user as User;
```

Comment:
a lot of repeats of that line. Can you perhaps override Request obj here - to extend it with user object